crates/iota-rest-api/src/response.rs

```diff
@@ -126,22 +127,57 @@ where
     }
 }
 
+#[derive(Debug)]
+pub enum HeaderError {
+    InternalServerError(String),
+}
```

Comment:
I am not very sure whether it is good to introduce a new `HeaderError`, or maybe you can just use the [`RestError` ](https://github.com/iotaledger/iota/blob/develop/crates/iota-rest-api/src/error.rs#L9-L12)defined in the `iota-rest-api` crate?